Fix DDLVector problem with boost 1.67

Dr15Jones opened the pull request:
Boost 1.67 had a chance to boost::spirit::classic which broke the parser in DDLVector.

Dr15Jones committed:
Removed the use of boost::spirit

With boost 1.67, the parser was no longer ignoring spaces at the
end of a Vector. Given the version of boost::spirit being used
was the deprecated one, it was replaced with a simple algorithm.

Dr15Jones committed:
Avoid unnecessary DDLGlobalRegistry calls

When parsing the elements of a Vector, we already know which
DDLVector to use (it is 'this').

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23592/5193

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones (Chris Jones) for master.

It involves the following packages:

DetectorDescription/Parser


@cmsbuild, @civanch, @Dr15Jones, @ianna, @mdhildreth can you please review it and eventually sign? Thanks.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23592/5194

cmsbuild commented:
Pull request #23592 was updated. @cmsbuild, @civanch, @Dr15Jones, @ianna, @mdhildreth can you please check and sign again.

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: b3d1dc8dc5d84ffac25407858ab4af556194edf5
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23592/28705/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23592/28705/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2902004
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2901812
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

Dr15Jones committed:
Avoid unnecessary DDLGlobalRegistry calls in DDLMap

Dr15Jones committed:
Removed the need for the DDLGlobalRegistry singleton

Now the DDLElementRegistry is owned by the DDLParser and passed
down to the helper classes which need it.

Dr15Jones committed:
Removed the use of a ClhepEvaluator singleton in DDLElementRegistry

The initialization of a ClhepEvaluator does not appear to be very
CPU intensive therefore there is no critical need to have only
one in the system.
This change should makes DDLElementRegistry thread-friendly.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23592/5203

cmsbuild commented:
Pull request #23592 was updated. @cmsbuild, @civanch, @Dr15Jones, @ianna, @mdhildreth can you please check and sign again.

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 07aa690dd05c84f3211c99d2df1435223de99c5d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23592/28715/summary.html

cmsbuild commented:
Comparison job queued.

Dr15Jones committed:
Removed the use of ClhepEvaluator as a Singleton

The ClhepEvaluator is now just used locally as part of the group
of objects which need it.

Dr15Jones committed:
Make sure DDConstants get filled

Once the expression pass has finished, we force the update of the
DDConstants from the ClhepEvaluator being used.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23592/5211

cmsbuild commented:
Pull request #23592 was updated. @cmsbuild, @civanch, @Dr15Jones, @ianna, @mdhildreth can you please check and sign again.

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: f94ca142344c8f78902256b61a0d5a7323899a0f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23592/28719/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23592/28719/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2902004
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2901813
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

civanch commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)
